src/Native/Unix/System.Native/pal_interfaceaddresses.cpp

```diff
@@ -44,6 +44,12 @@ extern "C" int32_t SystemNative_EnumerateInterfaceAddresses(IPv4AddressFound onI
     for (ifaddrs* current = headAddr; current != nullptr; current = current->ifa_next)
     {
         uint32_t interfaceIndex = if_nametoindex(current->ifa_name);
+        // ifa_name may be an aliased interface name.
+        // Use if_indextoname to map back to the true device name.
+        char actualName[IF_NAMESIZE];
+        char* result = if_indextoname(interfaceIndex, actualName);
+        assert(result == actualName && result != nullptr);
```

Comment:
Hm, that is a possibility. There's also the possibility that, during the enumeration of interfaces, system configuration changes. And by that, I mean in between enumeration of individual addresses (rather than in between `if_nametoindex` and `if_indextoname`). I haven't seen any code online that attempts to handle this, so I'm not sure what the best thing to do is.

For the specific case you mention, I wonder if it would be acceptable to return an error code and then throw a `PlatformNotSupportedException` from managed code? We could simply skip that address, but other addresses were probably already enumerated and may not be valid any longer.